packages/cmf/__tests__/cmfModule.merge.test.js

```diff
@@ -0,0 +1,192 @@
+import mergeModules from '../src/cmfModule.merge';
+
+describe('mergeModule', () => {
+	beforeEach(() => {
+		mergeModules.onOverride = jest.fn();
+	});
+	it('should merge components config', () => {
+		const a = {
+			components: {
+				foo: function foo() {},
+				bar: function bar() {},
+			},
+		};
+		const b = {
+			components: {
+				foo: function foo() {},
+			},
+		};
+		const config = mergeModules(a, b);
+		expect(config.components.foo).toBe(b.components.foo);
+		expect(config.components.bar).toBe(a.components.bar);
+		expect(config.components.foo).toBe(b.components.foo);
+		expect(config.components.bar).toBe(a.components.bar);
+	});
+	it('should merge sagas config', () => {
+		const a = {
+			sagas: {
+				foo: function* foo() {},
+				bar: function* bar() {},
+			},
+		};
+		const b = {
+			sagas: {
+				foo: function* foo() {},
+			},
+		};
+		const config = mergeModules(a, b);
+		expect(config.sagas.foo).toBe(b.sagas.foo);
+		expect(config.sagas.bar).toBe(a.sagas.bar);
+		expect(config.sagas.foo).toBe(b.sagas.foo);
+		expect(config.sagas.bar).toBe(a.sagas.bar);
+	});
+	it('should merge expressions config', () => {
+		const a = {
+			expressions: {
+				foo: function* foo() {},
+				bar: function* bar() {},
+			},
+		};
+		const b = {
+			expressions: {
+				foo: function* foo() {},
+			},
+		};
+		const config = mergeModules(a, b);
+		expect(config.expressions.foo).toBe(b.expressions.foo);
+		expect(config.expressions.bar).toBe(a.expressions.bar);
+		expect(config.expressions.foo).toBe(b.expressions.foo);
+		expect(config.expressions.bar).toBe(a.expressions.bar);
+	});
+	it('should merge actionCreators config', () => {
+		const a = {
+			actionCreators: {
+				foo: function* foo() {},
+				bar: function* bar() {},
+			},
+		};
+		const b = {
+			actionCreators: {
+				foo: function* foo() {},
+			},
+		};
+		const config = mergeModules(a, b);
+		expect(config.actionCreators.foo).toBe(b.actionCreators.foo);
+		expect(config.actionCreators.bar).toBe(a.actionCreators.bar);
+		expect(config.actionCreators.foo).toBe(b.actionCreators.foo);
+		expect(config.actionCreators.bar).toBe(a.actionCreators.bar);
+	});
+	it('should throw an exeption on unknown keys', () => {
+		const toThrow = () => mergeModules({ foo: {} });
+		expect(toThrow).toThrow();
+	});
+	it('should throw an exeption if two config has appId', () => {
+		const toThrow = () => mergeModules({ appId: 'foo' }, { appId: 'bar' });
+		expect(toThrow).toThrow();
+	});
+	it('should get first appId in config', () => {
```

Comment:
I don't think this test is really necessary. Or just test 1 case, where there is an app id in one of the configs and check that it is taken in the merged config.
The description is disturbing, as we only allow a unique app id